### PR TITLE
Collections: Refactor CRUD moving hyrax/collections to hyrax/dashboard/collections

### PR DESCRIPTION
**Pull request by elrayle**
Partially addresses #1380 Collections: Adjust routes to accommodate a public landing page and permission templates.

 Refactor CRUD moving hyrax/collections to hyrax/dashboard/collections.  Also adds a second show page under hyrax/dashcoard/collections that only those with edit access can view and use to perform management tasks for the collection.  The original show page under hyrax/collections has had management tasks removed in preparation of becoming the public branded landing page.
    
There are a few errors that were not fully resolved as part of this commit.  See...
* #1488 Collection CRUD refactor - test 'removes all works' fails
* #1489 Collection CRUD refactor - Throwing ActionController::InvalidAuthenticityToken when trying to add works to an existing collection

Impacts issues related to CRUD for Collections, including...
* #1417 Collections: Update UI for index views (All/Your) of MVC for Managing Collections
* #1427 Collections: Create an admin show view
* #1428 Collections: Update public landing page to include branding and other UI recommendations
* #1374 Collections: Modify existing new/edit view to include partials for new tabs

Primary changes that impact these issues...
* All issues...
  * many paths are updated to include dashboard in the path
* Effecting 1374 new/edit view
  * management views and partials (e.g. new, edit, + partials) moved to [views/app/hyrax/dashboard/collections](https://github.com/samvera/hyrax/tree/collections-sprint/app/views/hyrax/dashboard/collections)
  * [app/controllers/concerns/hyrax/collections_controller_behavior.rb](https://github.com/samvera/hyrax/blob/collections-sprint/app/controllers/concerns/hyrax/collections_controller_behavior.rb) no longer controls new/edit/destroy CRUD.
  * [app/controllers/concerns/hyrax/dashboard/collections_controller_behavior.rb](https://github.com/samvera/hyrax/blob/collections-sprint/app/controllers/concerns/hyrax/dashboard/collections_controller_behavior.rb) now controls new/edit/destroy CRUD.  NOTE: This controller behavior is included in [app/controllers/hyrax/dashboard/collections_controller.rb](https://github.com/samvera/hyrax/blob/collections-sprint/app/controllers/hyrax/dashboard/collections_controller.rb) which inherits from [app/controllers/hyrax/my/collections_controller.rb](https://github.com/samvera/hyrax/blob/collections-sprint/app/controllers/hyrax/my/collections_controller.rb)
* Effecting 1427 & 1428 both show pages
  * many of the show partials were copied to  [views/app/hyrax/dashboard/collections](https://github.com/samvera/hyrax/tree/collections-sprint/app/views/hyrax/dashboard/collections) to serve as the basis for the admin version of the show page
  * the show views and all partials remained in [views/app/hyrax/collections](https://github.com/samvera/hyrax/tree/collections-sprint/app/views/hyrax/collections) to serve as the basis for the public landing show page
  * many of the show partials are only in [views/app/hyrax/collections](https://github.com/samvera/hyrax/tree/collections-sprint/app/views/hyrax/collections) and are shared by the admin show page.  If these partials diverge for the two show pages, some may need to be copied to [views/app/hyrax/dashboard/collections](https://github.com/samvera/hyrax/tree/collections-sprint/app/views/hyrax/dashboard/collections) to accommodate the needs of the admin show page
  * [app/controllers/concerns/hyrax/collections_controller_behavior.rb](https://github.com/samvera/hyrax/blob/collections-sprint/app/controllers/concerns/hyrax/collections_controller_behavior.rb) continues to have the show related action code for the public branded landing page
  * [app/controllers/concerns/hyrax/dashboard/collections_controller_behavior.rb](https://github.com/samvera/hyrax/blob/collections-sprint/app/controllers/concerns/hyrax/dashboard/collections_controller_behavior.rb) has the show actions copied over.  NOTE: This controller behavior is included in [app/controllers/hyrax/dashboard/collections_controller.rb](https://github.com/samvera/hyrax/blob/collections-sprint/app/controllers/hyrax/dashboard/collections_controller.rb) which inherits from [app/controllers/hyrax/my/collections_controller.rb](https://github.com/samvera/hyrax/blob/collections-sprint/app/controllers/hyrax/my/collections_controller.rb)
* Effecting 1417 index views
  * There is not much change at all for the index views.  Changes will primarily include path changes to outgoing URLs for collection CRUD actions.

### CODE REVIEW

**Comment by elrayle** [comment]
**DO NOT MERGE**

File: app/controllers/concerns/hyrax/dashboard/collections_controller_behavior.rb [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+        # actions: index, create, new, edit, show, update, destroy, permissions, citation
+        before_action :authenticate_user!, except: [:index]
+
+        class_attribute :presenter_class,
+                        :form_class,
+                        :single_item_search_builder_class,
+                        :member_search_builder_class
+
+        alias_method :collection_search_builder_class, :single_item_search_builder_class
```
Comment (jeremyf) [inline app/controllers/concerns/hyrax/dashboard/collections_controller_behavior.rb]: Do you need an alias on the class methods as well? What about the setter methods?

File: app/controllers/hyrax/my_controller.rb [hunk 2]
```diff
@@ -22,8 +22,7 @@ def self.configure_facets
     configure_facets
 
     before_action :authenticate_user!
-    before_action :enforce_show_permissions, only: :show
-    before_action :enforce_viewing_context_for_show_requests, only: :show
+    # before_action :enforce_show_permissions, only: :show
```
Comment (jeremyf) [inline app/controllers/hyrax/my_controller.rb]: Remove commented out code.

File: app/views/hyrax/collections/_list_collections.html.erb [hunk 3]
```diff
@@ -6,7 +6,8 @@
       <span class="<%= Hyrax::ModelIcon.css_class_for(Collection) %> collection-icon-small pull-left"></span>
       <div class="media-body">
         <div class="media-heading">
-          <%= link_to url_for_document(document), id: "src_copy_link#{id}" do %>
+          <%#= link_to url_for_document(document), id: "src_copy_link#{id}" do %>
```
Comment (jeremyf) [inline app/views/hyrax/collections/_list_collections.html.erb]: Remove commented out code.

File: app/views/hyrax/dashboard/collections/_button_for_remove_selected_from_collection.html.erb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <%# button for removing a batch from a collection %>
 <%# collection -- collection to be updated  %>
 <%# label -- button label %>
-<%= form_for collection, url:collection_path(collection.id), :method=>:put do |f| %>
+<%= form_for collection, url: hyrax.dashboard_collection_path(collection.id), :method=>:put do |f| %>
```
Comment (jeremyf) [inline app/views/hyrax/dashboard/collections/_button_for_remove_selected_from_collection.html.erb]: Switch `:method => :put` to `method: :put`

File: app/views/hyrax/dashboard/collections/_list_collections.html.erb [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<% id = document.id %>
+<tr id="document_<%= id %>">
+  <td></td>
+  <td>
+    <div class="media">
+      <span class="<%= Hyrax::ModelIcon.css_class_for(Collection) %> collection-icon-small pull-left"></span>
+      <div class="media-body">
+        <div class="media-heading">
+          <%#= link_to url_for_document(document), id: "src_copy_link#{id}" do %>
```
Comment (hortongn) [inline app/views/hyrax/dashboard/collections/_list_collections.html.erb]: Remove commented code

File: app/views/hyrax/my/collections/_list_collections.html.erb [hunk 6]
```diff
@@ -5,7 +5,8 @@
       <span class="<%= Hyrax::ModelIcon.css_class_for(Collection) %> collection-icon-small pull-left"></span>
       <div class="media-body">
         <div class="media-heading">
-          <%= link_to url_for_document(document), id: "src_copy_link#{id}" do %>
+          <%#= link_to url_for_document(document), id: "src_copy_link#{id}" do %>
```
Comment (hortongn) [inline app/views/hyrax/my/collections/_list_collections.html.erb]: Remove commented code

**Comment by hortongn** [comment]
@elrayle I didn't run it, but I looked through the code.  Other than a couple more commented lines of code that should be removed, I don't have any other changes beyond what Jeremy suggested.

**Comment by jeremyf** [comment]
@elrayle Should this be closed?

**Comment by elrayle** [comment]
This was addressed by the master PR #1499 which was included in the collections-sprint branch through a rebase with master last night.

**Pull request closed without merging**